### PR TITLE
(PIE-656) Temp Files for Processors

### PR DESCRIPTION
**Pull request by RandomNoun7**
This change implements writing temp files for each processor that gets
invoked.

### CODE REVIEW

File: spec/unit/util/common_events/processor_spec.rb [hunk 1]
```diff
@@ -33,7 +40,7 @@
         end
 
         it 'returns correct object types' do
-          described_class.find_each(procs_dir).each do |processor|
+          described_class.find_each(procs_dir) do |processor|
```
Comment (nam054) [inline spec/unit/util/common_events/processor_spec.rb:43]: Possible conflict with the subject processor?

**Review by nam054 — CHANGES_REQUESTED** [review]
Just a few questions

File: spec/support/acceptance/proc1.sh [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #! /bin/bash
 
-echo 'hello world!' > /tmp/proc1
+cat $1 > /tmp/proc1.out
```
[the remark this answers is not in the document]
Comment (RandomNoun7) [reply spec/support/acceptance/proc1.sh:3]: Good point. Done.

File: files/collect_api_events.rb [hunk 3]
```diff
@@ -40,7 +43,11 @@ def main(confdir, _modulepaths, statedir)
       index.save(service => current_count)
     end
 
-    puts data.to_json
+    CommonEvents::Processor.find_each("#{confdir}/processors.d") do |processor|
+      processor.invoke(data)
+      log.info(processor.stdout, source: processor.name)
+      log.warn(processor.stderr, source: processor.name) unless processor.stderr.nil?
```
[the remark this answers is not in the document]
Comment (RandomNoun7) [reply files/collect_api_events.rb:49]: Turns out, testing exit code as a proxy for whether there's anything in `stderr` is a bad idea.

File: spec/unit/util/common_events/processor_spec.rb (hunk 1 above)
Comment (RandomNoun7) [reply spec/unit/util/common_events/processor_spec.rb:43]: Good catch. Name changed to avoid conflict.

**Pull request merged**